pkg/controller/seed-controller-manager/addon/addon_controller_test.go

```diff
@@ -418,7 +446,57 @@ func TestHugeManifest(t *testing.T) {
 		kubernetesAddonDir: "./testdata",
 		KubeconfigProvider: &fakeKubeconfigProvider{},
 	}
-	if _, _, _, err := r.setupManifestInteraction(context.Background(), log, addon, cluster); err != nil {
-		t.Fatalf("failed to setup manifest interaction: %v", err)
+	if mi := r.setupManifestInteraction(context.Background(), log, addon, cluster); mi.err != nil {
+		t.Fatalf("failed to setup manifest interaction: %v", mi.err)
+	}
+}
+
+func TestController_getAddonCSICinderServiceSeed(t *testing.T) {
+	cluster := setupTestCluster("10.240.16.0/20")
+	clusterNS := "cluster-7xkdhv595z"
+	cluster.Status.NamespaceName = clusterNS
+	addon := setupTestAddon("kube-dns")
+
+	addonDir, err := ioutil.TempDir("/tmp", "kubermatic-tests-")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(addonDir)
+
+	if err := os.Mkdir(path.Join(addonDir, addon.Spec.Name), 0777); err != nil {
+		t.Fatal(err)
+	}
+	if err := ioutil.WriteFile(path.Join(addonDir, addon.Spec.Name, "testManifest.yaml"), []byte(testManifestForSeed), 0644); err != nil {
+		t.Fatal(err)
+	}
+
+	log := kubermaticlog.New(true, kubermaticlog.FormatConsole).Sugar()
+	ctx := context.Background()
+
+	controller := &Reconciler{
+		kubernetesAddonDir: addonDir,
+		KubeconfigProvider: &fakeKubeconfigProvider{},
+	}
+	manifests, err := controller.getAddonManifests(ctx, log, addon, cluster, templateAddonForSeedCluster)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(manifests) != 1 {
+		t.Fatalf("invalid number of manifests returned. Expected 1, Got %d", len(manifests))
+	}
+	svc := v1.Service{}
+	if err := json.Unmarshal(manifests[0].Raw, &svc); err != nil {
+		t.Fatalf("failed to unmarshal raw service: %v", err)
+	}
+	if svc.Namespace != cluster.Status.NamespaceName || svc.Namespace != clusterNS {
+		t.Fatalf("svc.Namespace[%v] != cluster.Status.NamespaceName[%v] != clusterNS[%v]", svc.Namespace, cluster.Status.NamespaceName, clusterNS)
+	}
+
+	manifests, err = controller.getAddonManifests(ctx, log, addon, cluster, templateAddonForUserCluster)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(manifests) != 0 {
```

Comment:
because the entire manifest is encapsulated within `{{ if eq .AddonCluster "seed" }}`, templating of the addon for user cluster should return no raw resources.